### PR TITLE
build: start build on z/OS

### PR DESCRIPTION
**Pull request by alexcfyung**
Same as #12, it was closed because of github bug that changes doesn't update to the PR.

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -842,6 +848,9 @@ else
 ifeq ($(findstring s390x,$(UNAME_M)),s390x)
 DESTCPU ?= s390x
 else
+ifeq ($(findstring OS/390,$(shell uname -s)),OS/390)
+DESTCPU ?= s390x
+else
```
Comment (joransiu) [inline Makefile]: Curious, why this is being added between s390x and s390?

File: common.gypi [hunk 2]
```diff
@@ -419,7 +427,7 @@
             'cflags': [ '-m64', '-mminimal-toc' ],
             'ldflags': [ '-m64' ],
           }],
-          [ 'target_arch=="s390x"', {
+          [ 'target_arch=="s390x" and OS!="zos"', {
```
Comment (joransiu) [inline common.gypi]: Perhaps we should check OS == "linux" instead here, as the options are LoZ specific.

File: configure.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 import shutil
 import bz2
 import io
+import tempfile
```
Comment (zsw007) [inline configure.py]:
```
./configure.py:16:1: F401 'tempfile' imported but unused
import tempfile
^
1     F401 'tempfile' imported but unused
1
```


```suggestion
```

File: Makefile (hunk 1 above)
Comment (alexcfyung) [reply Makefile]: I don't think there is a particular reason, @gabylb can you confirm? I can move it below s390 if that is more preferable

Comment (joransiu) [reply Makefile]: Yeah, I'd think that's more logical.

File: configure.py [hunk 4]
```diff
@@ -1051,7 +1066,10 @@ def is_arm_hard_float_abi():
 def host_arch_cc():
   """Host architecture check using the CC command."""
 
-  k = cc_macros(os.environ.get('CC_host'))
+  if sys.platform.startswith('zos'):
+    return 's390x';
+  else:
+    k = cc_macros(os.environ.get('CC_host'))
```
Comment (zsw007) [inline configure.py]: The `else` seems to be redundant.

```suggestion
  k = cc_macros(os.environ.get('CC_host'))
```

File: configure.py [hunk 5]
```diff
@@ -1051,6 +1065,8 @@ def is_arm_hard_float_abi():
 def host_arch_cc():
   """Host architecture check using the CC command."""
 
+  if sys.platform.startswith('zos'):
+    return 's390x';
```
Comment (zsw007) [inline configure.py]: Accidental semicolon

```suggestion
    return 's390x'
```

**Review by joransiu — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (gabylb) [reply Makefile]: Probably to group the two `DESTCPU ?= s390x` cases.

**Review by gabylb — APPROVED** [review]
LGTM

**Review by zsw007 — APPROVED** [review]
LGTM

**Comment by zsw007** [comment]
Are all 13 commits intended to be upstreamed? I feel like we might be able to squash some of them.

**Comment by alexcfyung** [comment]
landed on https://github.com/nodejs/node/commit/f6e5050579d8d0ec7c68ed499dcc8701bd49c18a

**Pull request merged**